pkg/handler/processor/process/process.go

```diff
@@ -251,3 +271,31 @@ func decompressDocument(i *processor.Document, reader io.Reader) error {
 	i.Blob = uncompressed
 	return nil
 }
+
+func detectFileEncoding(i *processor.Document) (string, error) {
+
+	//create a bufio.Reader so we can 'peek' at the first few bytes without consuming
+	bReader := bytes.NewReader(i.Blob)
+	reader := bufio.NewReader(bReader)
+	blankType := ""
+	if len(i.Blob) >= 16 {
+		testBytes, err := reader.Peek(16)
+		if err != nil {
+			return "", err
+		}
+
+		//find the content mime-type from the first few bytes
+		contentType := http.DetectContentType(i.Blob)
+		//octet-stream is the default meaning that no encoding has been found
+		if contentType == "application/octet-stream" {
+			if testBytes[0] == 0x42 && testBytes[1] == 0x5A && testBytes[2] == 0x68 {
```

Comment:
nit: can we create a function for checking the header and maybe add a link out/comment on what they check is doing (i.e. checking the file header, according to RFC xyz).